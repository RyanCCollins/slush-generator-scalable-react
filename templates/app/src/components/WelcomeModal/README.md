## WelcomeModal Component
A component that ...

### Example

```js
<WelcomeModal />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
