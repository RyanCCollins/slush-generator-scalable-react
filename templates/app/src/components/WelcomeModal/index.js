import React, { PropTypes } from 'react';
import Layer from 'grommet-udacity/components/Layer';
import Form from 'grommet-udacity/components/Form';
import FormFields from 'grommet-udacity/components/FormFields';
import FormField from 'grommet-udacity/components/FormField';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import Button from 'grommet-udacity/components/Button';
import Footer from 'grommet-udacity/components/Footer';
import { Divider } from 'components';
import error from './utils/error';

const WelcomeModal = ({
  onClose,
  isVisible,
  nameInput,
  onSubmit,
}) => (
  <Layer
    align="center"
    closer
    hidden={!isVisible}
    onClose={onClose}
  >
    <Box align="center" justify="center" pad="large">
      <Heading align="center">
        What should we call you?
      </Heading>
      <Divider />
      <Form>
        <FormFields>
          <FormField
            help="What should we call you?"
            error={error(nameInput)}
            label="Enter your name"
            htmlFor="nameInput"
          >
            <input
              {...nameInput}
              required
              autoFocus
              placeholder="Ryan Collins"
              id="nameInput"
              autoComplete="on"
              name="name"
              type="text"
              aria-invalid={nameInput.error}
              aria-required
              className="input"
            />
          </FormField>
        </FormFields>
      </Form>
      <Footer align="center" justify="center" pad="large">
        <Button primary label="Submit" onClick={onSubmit} />
      </Footer>
    </Box>
  </Layer>
);

WelcomeModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  nameInput: PropTypes.shape({
    error: PropTypes.string,
    value: PropTypes.string,
  }),
  isVisible: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default WelcomeModal;
