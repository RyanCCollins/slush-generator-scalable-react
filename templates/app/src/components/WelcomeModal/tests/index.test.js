import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import fields from './mocks';
import WelcomeModal from '../index';

describe('<WelcomeModal />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <WelcomeModal
        isVisible={false}
        onClose={e => e}
        onSubmit={e => e}
        {...fields}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
