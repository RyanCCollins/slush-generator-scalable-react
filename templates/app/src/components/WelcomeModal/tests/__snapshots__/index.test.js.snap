exports[`<WelcomeModal /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true}
  onClose={[Function]}>
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="large"
    responsive={true}
    tag="div">
    <Heading
      align="center"
      tag="h1">
      What should we call you?
    </Heading>
    <undefined />
    <Form
      compact={false}
      fill={false}
      pad="none">
      <FormFields>
        <FormField
          error="Value Required"
          help="What should we call you?"
          htmlFor="nameInput"
          label="Enter your name"
          size="medium"
          strong={false}>
          <input
            active={false}
            aria-invalid="Value Required"
            aria-required={true}
            autoComplete="on"
            autoFocus={true}
            autofilled={false}
            className="input"
            dirty={false}
            error="Value Required"
            id="nameInput"
            initialValue=""
            invalid={true}
            name="name"
            placeholder="Ryan Collins"
            pristine={true}
            required={true}
            touched={true}
            type="text"
            valid={false}
            value=""
            visited={true} />
        </FormField>
      </FormFields>
    </Form>
    <Footer
      align="center"
      direction="row"
      justify="center"
      pad="large"
      responsive={false}>
      <Button
        label="Submit"
        onClick={[Function]}
        primary={true}
        type="button" />
    </Footer>
  </Box>
</Layer>
`;
